network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -156,21 +164,22 @@ func DefaultResourceManagerConfig() *ResourceManagerConfig {
 }
 
 type LibP2PNodeBuilder struct {
-	sporkID              flow.Identifier
-	addr                 string
-	networkKey           fcrypto.PrivateKey
-	logger               zerolog.Logger
-	metrics              module.LibP2PMetrics
-	basicResolver        madns.BasicResolver
-	subscriptionFilter   pubsub.SubscriptionFilter
-	resourceManager      network.ResourceManager
-	resourceManagerCfg   *ResourceManagerConfig
-	connManager          connmgr.ConnManager
-	connGater            connmgr.ConnectionGater
-	idProvider           module.IdentityProvider
-	gossipSubFactory     GossipSubFactoryFunc
-	gossipSubConfigFunc  GossipSubAdapterConfigFunc
-	gossipSubPeerScoring bool // whether to enable gossipsub peer scoring
+	sporkID                      flow.Identifier
+	addr                         string
+	networkKey                   fcrypto.PrivateKey
+	logger                       zerolog.Logger
+	metrics                      module.LibP2PMetrics
+	basicResolver                madns.BasicResolver
+	subscriptionFilter           pubsub.SubscriptionFilter
+	resourceManager              network.ResourceManager
+	resourceManagerCfg           *ResourceManagerConfig
+	connManager                  connmgr.ConnManager
+	connGater                    connmgr.ConnectionGater
+	idProvider                   module.IdentityProvider
+	gossipSubFactory             GossipSubFactoryFunc
+	gossipSubConfigFunc          GossipSubAdapterConfigFunc
+	gossipSubPeerScoring         bool          // whether to enable gossipsub peer scoring
+	gossipSubScoreTracerInterval time.Duration // the interval at which the gossipsub score tracer logs the peer scores.
 
 	// gossipSubTracer is a callback interface that is called by the gossipsub implementation upon
```

Comment:
Should we combine these into the current GossipSubConfig struct or some other struct that encapsulates all the different gossipsub configurations. It seems like there may be more gossipsub specific configurations in the future.

Reply:
https://github.com/onflow/flow-go/pull/3986/commits/9deed66acb5beba3c679d358ab57b69ea45bb461